runtime/stdlib/account.go

```diff
@@ -1267,7 +1277,85 @@ func newAccountContractsGetFunction(
 				return interpreter.Nil
 			}
 		},
-		sema.AuthAccountContractsTypeGetFunctionType,
+		sema.AccountContractsTypeGetFunctionType,
+	)
+}
+
+func newAccountContractsBorrowFunction(
+	gauge common.MemoryGauge,
+	handler PublicAccountContractsHandler,
+	addressValue interpreter.AddressValue,
+) *interpreter.HostFunctionValue {
+
+	// Converted addresses can be cached and don't have to be recomputed on each function invocation
+	address := addressValue.ToAddress()
+
+	return interpreter.NewHostFunctionValue(
+		gauge,
+		func(invocation interpreter.Invocation) interpreter.Value {
+
+			inter := invocation.Interpreter
+
+			nameValue, ok := invocation.Arguments[0].(*interpreter.StringValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+			name := nameValue.Str
+
+			typeParameterPair := invocation.TypeParameterTypes.Oldest()
+			if typeParameterPair == nil {
+				panic(errors.NewUnreachableError())
+			}
+			ty := typeParameterPair.Value
+
+			referenceType, ok := ty.(*sema.ReferenceType)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+
+			// Check if the contract exists
+
+			var code []byte
+			var err error
+			wrapPanic(func() {
+				code, err = handler.GetAccountContractCode(address, name)
+			})
+			if err != nil {
+				panic(err)
+			}
+			if len(code) == 0 {
+				return interpreter.Nil
+			}
+
+			// Load the contract
+
+			contractLocation := common.NewAddressLocation(gauge, address, name)
+			subInterpreter := inter.EnsureLoaded(contractLocation)
+			contractValue, err := subInterpreter.GetContractComposite(contractLocation)
```

Comment:
For my own understanding, why do we use the subInterpreter here but not anywhere else later in this function?

Reply:
@dsainati1 I don't know really, I confess I did little cargo cult programming here  :( maybe @turbolent can explain

Reply:
We should use `subInterpreter`, I missed that. Thanks for the flag!

Reply:
Fixed in [e27fe5a](https://github.com/onflow/cadence/pull/1934/commits/e27fe5a8b768249f67650646b44b2f74e73e4c07)

Reply:
@turbolent any chance you can explain the reason? Now I looked up the place I borrowed the code [0], a bit confused. 

[0] https://github.com/onflow/cadence/blob/2926bebcebb6de2a8e3e4ddf148e4ed84967c2d6/runtime/interpreter/interpreter_import.go#L40

Reply:
I guess the reason here is `ConvertStaticToSemaType` (called deep within `IsSubTypeOfSemaType`) uses the elaboration from the interpreter to load types. So if we don't use the imported-interpreter (aka subInterpreter), It will fail to load the type of the imported/loaded contract during runtime subtype validation.